src/comment_ast.rs

```diff
@@ -92,32 +104,45 @@ pub fn metadata_from_comments(comments: &[&str]) -> RuleMetadata {
 
 #[test]
 fn parse_comment_name() {
-  assert_eq!(rule_metadata("@name foo"), Ok(("", RuleMetadata {
-    name: Some("foo".to_string()),
-    is_newtype: false,
-  })));
+    assert_eq!(rule_metadata("@name foo"), Ok(("", RuleMetadata {
+        name: Some("foo".to_string()),
+        is_newtype: false,
+        no_alias: false,
+    })));
 }
 
 #[test]
 fn parse_comment_newtype() {
-  assert_eq!(rule_metadata("@newtype"), Ok(("", RuleMetadata {
-    name: None,
-    is_newtype: true
-  })));
+    assert_eq!(rule_metadata("@newtype"), Ok(("", RuleMetadata {
+        name: None,
+        is_newtype: true,
+        no_alias: false,
+    })));
 }
 
 #[test]
 fn parse_comment_newtype_and_name() {
-  assert_eq!(rule_metadata("@newtype @name foo"), Ok(("", RuleMetadata {
-    name: Some("foo".to_string()),
-    is_newtype: true
-  })));
+    assert_eq!(rule_metadata("@newtype @name foo"), Ok(("", RuleMetadata {
+        name: Some("foo".to_string()),
+        is_newtype: true,
+        no_alias: false,
+    })));
 }
 
 #[test]
 fn parse_comment_newtype_and_name_inverse() {
-  assert_eq!(rule_metadata("@name foo @newtype"), Ok(("", RuleMetadata {
-    name: Some("foo".to_string()),
-    is_newtype: true
-  })));
-}
\ No newline at end of file
+    assert_eq!(rule_metadata("@name foo @newtype"), Ok(("", RuleMetadata {
+        name: Some("foo".to_string()),
+        is_newtype: true,
+        no_alias: false,
+    })));
+}
+
+#[test]
+fn parse_comment_all() {
+    assert_eq!(rule_metadata("@no_alias @name foo @newtype"), Ok(("", RuleMetadata {
```

Comment:
Does using both `no_alias` and `newtype` at the same time generate the right code?

Reply:
Perhaps it's best to error here? If you use `@newtype` then there would never be an alias in the first place. Should we make `@no_alias` be effectively a no-op when you use `@newtype`? I think one of those two options is best, leaning slightly more towards the error since I don't see why you would want to apply both.

Reply:
It errors now. I put in a test to make sure it panics as well.